### PR TITLE
Replace ifdefs with hardware abstraction layer

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

File: selfdrive/camerad/cameras/camera_common.cc [hunk 1]
```diff
@@ -350,21 +347,27 @@ static void driver_cam_auto_exposure(CameraState *c, SubMaster &sm) {
   static const bool is_rhd = Params().getBool("IsRHD");
   struct ExpRect {int x1, x2, x_skip, y1, y2, y_skip;};
   const CameraBuf *b = &c->buf;
-#ifndef QCOM2
+
   bool hist_ceil = false, hl_weighted = false;
+  int x_offset = 0, y_offset = 0;
+  int frame_width = b->rgb_width, frame_height = b->rgb_height;
+#ifndef QCOM2
```
Comment (pd0wm) [inline selfdrive/camerad/cameras/camera_common.cc:354]: Still an ifdef here

File: selfdrive/loggerd/logger.h [hunk 2]
```diff
@@ -8,13 +8,11 @@
...
+#include "selfdrive/hardware/hw.h"
 
-#if defined(QCOM) || defined(QCOM2)
-const std::string LOG_ROOT = "/data/media/0/realdata";
-#else
-const std::string LOG_ROOT = util::getenv_default("HOME", "/.comma/media/0/realdata", "/data/media/0/realdata");
-#endif
-
+const std::string LOG_ROOT =
+    !Hardware::PC() ? "/data/media/0/realdata"
```
Comment (pd0wm) [inline selfdrive/loggerd/logger.h:14]: No need for negative here, just swap the values.

File: selfdrive/loggerd/loggerd.cc [hunk 3]
```diff
@@ -369,18 +359,16 @@ int main(int argc, char** argv) {
   encoder_threads.push_back(std::thread(encoder_thread, LOG_CAMERA_ID_FCAMERA));
   s.rotate_state[LOG_CAMERA_ID_FCAMERA].enabled = true;
 
-#if defined(QCOM) || defined(QCOM2)
-  bool record_front = Params().getBool("RecordFront");
-  if (record_front) {
-    encoder_threads.push_back(std::thread(encoder_thread, LOG_CAMERA_ID_DCAMERA));
-    s.rotate_state[LOG_CAMERA_ID_DCAMERA].enabled = true;
+  if (!Hardware::PC()) {
```
Comment (pd0wm) [inline selfdrive/loggerd/loggerd.cc:362]: Combine with next if

File: selfdrive/modeld/models/dmonitoring.cc [hunk 4]
```diff
@@ -55,24 +51,25 @@ void crop_yuv(uint8_t *raw, int width, int height, uint8_t *y, uint8_t *u, uint8
...
-  const int adapt_width_tici = 668;
-  const int cropped_height = adapt_width_tici / 1.33;
-  Rect crop_rect = {full_width_tici / 2 - adapt_width_tici / 2,
-                    full_height_tici / 2 - cropped_height / 2 - 196,
-                    cropped_height / 2,
-                    cropped_height};
-  if (!s->is_rhd) {
-    crop_rect.x += adapt_width_tici - crop_rect.w + 32;
+  Rect crop_rect;
+  if (!Hardware::TICI()) {
```
Comment (pd0wm) [inline selfdrive/modeld/models/dmonitoring.cc:55]: Remove negative and swap with else

File: selfdrive/ui/main.cc [hunk 5]
```diff
@@ -17,9 +17,6 @@ int main(int argc, char *argv[]) {
 
 #ifdef QCOM
   QApplication::setAttribute(Qt::AA_ShareOpenGLContexts);
-#endif
-
-#ifdef QCOM
```
Comment (pd0wm) [inline selfdrive/ui/main.cc:22]: Why still an ifdef?

File: selfdrive/ui/paint.cc [hunk 6]
```diff
@@ -246,13 +242,9 @@ static void ui_draw_driver_view(UIState *s) {
...
   // blackout
   const int blackout_x_r = valid_rect.right();
-#ifndef QCOM2
-  const int blackout_w_r = rect.right() - valid_rect.right();
-  const int blackout_x_l = rect.x;
-#else
-  const int blackout_w_r = s->viz_rect.right() - valid_rect.right();
-  const int blackout_x_l = s->viz_rect.x;
-#endif
+  const Rect &blackout_rect = Hardware::TICI() ? rect : s->viz_rect;
```
Comment (pd0wm) [inline selfdrive/ui/paint.cc:245]: Think the condition is swapped here

File: selfdrive/ui/paint.cc [hunk 7]
```diff
@@ -479,41 +471,42 @@ static const mat4 device_transform = {{
...
-  xscale,  0.0, 0.0, xscale*crop_x_offset/full_width_tici*2,
-  0.0,  yscale, 0.0, yscale*crop_y_offset/full_height_tici*2,
-  0.0,  0.0, 1.0, 0.0,
-  0.0,  0.0, 0.0, 1.0,
-}};
-#endif
+static mat4 get_driver_view_transform() {
+  const float driver_view_ratio = 1.333;
+  mat4 transform;
+  if (!Hardware::TICI()) {
```
Comment (pd0wm) [inline selfdrive/ui/paint.cc:477]: Invert condition and swap with else.

File: selfdrive/ui/paint.cc [hunk 8]
```diff
@@ -479,41 +471,42 @@ static const mat4 device_transform = {{
...
   // init drawing
-#ifdef QCOM
-  // on QCOM, we enable MSAA
-  s->vg = nvgCreate(0);
-#else
-  s->vg = nvgCreate(NVG_ANTIALIAS | NVG_STENCIL_STROKES | NVG_DEBUG);
-#endif
+
+  // on EON, we enable MSAA
+  s->vg = Hardware::EON() ? nvgCreate(0) : s->vg = nvgCreate(NVG_ANTIALIAS | NVG_STENCIL_STROKES | NVG_DEBUG);
```
Comment (pd0wm) [inline selfdrive/ui/paint.cc:509]:
```suggestion
  s->vg = Hardware::EON() ? nvgCreate(0) : nvgCreate(NVG_ANTIALIAS | NVG_STENCIL_STROKES | NVG_DEBUG);
```

File: selfdrive/ui/qt/api.cc [hunk 9]
```diff
@@ -13,12 +13,11 @@
...
 #include "common/util.h"
+#include "selfdrive/hardware/hw.h"
 
-#if defined(QCOM) || defined(QCOM2)
-const std::string private_key_path = "/persist/comma/id_rsa";
-#else
-const std::string private_key_path = util::getenv_default("HOME", "/.comma/persist/comma/id_rsa", "/persist/comma/id_rsa");
-#endif
+const std::string private_key_path =
+    !Hardware::PC() ? "/persist/comma/id_rsa"
```
Comment (pd0wm) [inline selfdrive/ui/qt/api.cc:19]: Remove negative and swap

File: selfdrive/ui/ui.cc [hunk 10]
```diff
@@ -192,11 +192,12 @@ static void update_state(UIState *s) {
   }
   if (sm.updated("sensorEvents")) {
     for (auto sensor : sm["sensorEvents"].getSensorEvents()) {
-      if (sensor.which() == cereal::SensorEventData::LIGHT) {
-#ifndef QCOM2
-        scene.light_sensor = sensor.getLight();
-#endif
-      } else if (!scene.started && sensor.which() == cereal::SensorEventData::ACCELERATION) {
+      if (!Hardware::TICI()) {
```
Comment (pd0wm) [inline selfdrive/ui/ui.cc:195]: Combine ifs

Comment (pd0wm) [inline selfdrive/ui/ui.cc:195]: Combine ifs

File: selfdrive/camerad/cameras/camera_common.cc (hunk 1 above)
Comment (deanlee) [reply selfdrive/camerad/cameras/camera_common.cc:354]: analog_gain is only declared in the header file of qcom2,maybe we can fix this next time

**Comment by deanlee** [comment]
An error occurred while compiling dmonitoring.cc: 
```
In file included from ./selfdrive/hardware/eon/hardware.h:6:
In file included from phonelibs/android_frameworks_native/include/gui/ISurfaceComposer.h:23:
In file included from phonelibs/android_system_core/include/utils/RefBase.h:20:
In file included from phonelibs/android_system_core/include/cutils/atomic.h:22:
In file included from /system/comma/usr/lib/clang/8.0.0/include/stdatomic.h:31:
/data/data/com.termux/files/usr/include/stdatomic.h:238:42: error: reference to 'memory_order' is ambiguous
static __inline void atomic_thread_fence(memory_order __order __attribute__((unused))) {
```

**Pull request merged**